engines/stark/ui/world/fmvscreen.cpp

```diff
@@ -54,13 +54,30 @@ FMVScreen::~FMVScreen() {
 	delete _surfaceRenderer;
 }
 
-void FMVScreen::play(Common::String name) {
+void FMVScreen::play(const Common::String &name) {
+	Common::SeekableReadStream *stream;
+
+	// Play the low-resolution video, if possible
 	if (!StarkSettings->getBoolSetting(Settings::kHighFMV) && StarkSettings->hasLowResFMV()) {
-		name.erase(name.size() - 4);
-		name += "_lo_res.bbb";
+		Common::String lowResName = name;
+		lowResName.erase(lowResName.size() - 4);
+		lowResName += "_lo_res.bbb";
+
+		stream = StarkArchiveLoader->getExternalFile(lowResName, "Global/");
+		if (stream) {
+			_decoder->loadStream(stream);
```

Comment:
Please improve the code organization so this section is not duplicated with the high-resolution variant.